app/actions/client.py

```diff
@@ -23,10 +24,14 @@ class Config:
 
 class FarmLocation(pydantic.BaseModel):
     location: tuple[float, float] = pydantic.Field(alias='_location')
-    time: str = pydantic.Field(alias='_time')
+    time: datetime = pydantic.Field(alias='_time')
     device_name: str
     official_tag: str
 
+    @pydantic.validator('time', pre=True, always=True)
+    def parse_time_string(cls, v):
+        return datetime.strptime(v, "%Y-%m-%dT%H:%M:%S.%fZ").replace(tzinfo=timezone.utc)
```

Comment:
Given that format, you can just let this parse as a datetime. Drop the `pre` argument and just have it ensure the timezone is set.